src/OpenSheetMusicDisplay/OpenSheetMusicDisplay.ts

```diff
@@ -13,14 +13,20 @@ import {MXLHelper} from "../Common/FileIO/Mxl";
 import {Promise} from "es6-promise";
 import {AJAX} from "./AJAX";
 import * as log from "loglevel";
+import { DrawingParametersEnum, DrawingParameters } from "../MusicalScore/Graphical/DrawingParameters";
 
 export class OpenSheetMusicDisplay {
     /**
      * The easy way of displaying a MusicXML sheet music file
      * @param container is either the ID, or the actual "div" element which will host the music sheet
      * @autoResize automatically resize the sheet to full page width on window resize
      */
-    constructor(container: string|HTMLElement, autoResize: boolean = false, backend: string = "svg") {
+    constructor(container: string|HTMLElement,
+                options: OSMDOptions = {
```

Comment:
Maybe put the creation of the defaults in a static function. There will be tons more options I think and the constructor will get overblown.

Reply:
yes, definitely should be somewhere else.
doesn't need to be static though, could also be a private function called in the constructor.

Reply:
Not sure about TS but in other languages accessing a private method before the constructor doesn't work

Reply:
It doesn't need to be before the constructor, i'd call it in the constructor.